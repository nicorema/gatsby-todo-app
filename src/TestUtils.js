import React from 'react'
import { IntlContextProvider, IntlProvider } from 'gatsby-plugin-intl'

export const renderWithProviders = component => (
  <IntlContextProvider value={{ language: 'en-US', routed: true }}>
    <IntlProvider locale="en">{component}</IntlProvider>
  </IntlContextProvider>
)
