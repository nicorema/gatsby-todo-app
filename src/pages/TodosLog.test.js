import React from 'react'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import { renderWithProviders } from '../TestUtils'

import TodoLog from './todos-log'

const mockStore = configureStore([])

describe('Todos Log Page', () => {
  let store
  let todoLogCmp

  beforeEach(() => {
    store = mockStore({
      app: { todos: [] },
    })

    todoLogCmp = renderer.create(
      renderWithProviders(
        <Provider store={store}>
          <TodoLog />
        </Provider>
      )
    )
  })

  it('renders correctly', () => {
    const snapShot = todoLogCmp.toJSON()
    expect(snapShot).toMatchSnapshot()
  })

  it('renders a <Map />', () => {
    const map = todoLogCmp.root.findByProps({ 'data-testid': 'map' })
    expect(map).toBeTruthy()
  })
})
