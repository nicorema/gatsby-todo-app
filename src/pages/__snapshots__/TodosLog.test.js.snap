// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todos Log Page renders correctly 1`] = `
<div
  className="css-1mjppp"
>
  <header
    className="sc-dlnjPT eSGkCO"
  >
    <h1>
      Todos Log
    </h1>
  </header>
  <div>
    <div
      className="sc-hKFyIo fwSFRg"
    >
      <div
        onClick={[Function]}
        onMouseDownCapture={[Function]}
        onMouseMove={[Function]}
        style={
          Object {
            "height": "100%",
            "margin": 0,
            "padding": 0,
            "position": "relative",
            "width": "100%",
          }
        }
      >
        <div
          style={
            Object {
              "height": "100%",
              "left": 0,
              "margin": 0,
              "padding": 0,
              "position": "absolute",
              "top": 0,
              "width": "100%",
            }
          }
        />
        <div
          style={
            Object {
              "height": "50%",
              "left": "50%",
              "margin": 0,
              "padding": 0,
              "position": "absolute",
              "top": "50%",
              "width": "50%",
            }
          }
        >
          <div
            style={
              Object {
                "height": "100%",
                "left": 0,
                "margin": 0,
                "padding": 0,
                "position": "absolute",
                "top": 0,
                "width": "100%",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
  <footer
    className="css-1ty9u91"
  >
    <nav
      className="sc-bdnylx giuwNz"
    >
      <ul>
        <li>
          <span
            className="css-qyfojd"
          >
            <a
              href="/todos-manager"
            >
              Todos Manager
            </a>
          </span>
        </li>
        <li>
          <span
            className="css-qyfojd"
          >
            <a
              href="/todos-log"
            >
              Todos Log
            </a>
          </span>
        </li>
      </ul>
    </nav>
  </footer>
</div>
`;
